File: s-personas/src/main/java/co/edu/javeriana/arqui/rest/ejb/ServicioPacientes.java
package co.edu.javeriana.arqui.rest.ejb;

import co.edu.javeriana.arqui.rest.entity.Paciente;
import co.edu.javeriana.arqui.rest.repository.RepositorioPacientes;
import jakarta.ejb.EJB;
import jakarta.ejb.Stateless;
import java.util.List;

@Stateless
public class ServicioPacientes {

    @EJB
    private RepositorioPacientes repoPacientes;

    public Paciente registrarPaciente(Paciente paciente) {
        repoPacientes.crear(paciente);
        return paciente;
    }

    public List<Paciente> listarPacientes() {
        return repoPacientes.listarTodos();
    }
}

File: s-personas/src/main/java/co/edu/javeriana/arqui/rest/ejb/ServicioPacientes.kt
package co.edu.javeriana.arqui.rest.ejb

import co.edu.javeriana.arqui.rest.entity.Paciente
import co.edu.javeriana.arqui.rest.repository.RepositorioPacientes
import jakarta.ejb.EJB
import jakarta.ejb.Stateless

@Stateless
open class ServicioPacientes {

    @EJB
    private lateinit var repoPacientes: RepositorioPacientes

    fun registrarPaciente(paciente: Paciente): Paciente {
        repoPacientes.crear(paciente)
        return paciente  // Devuelve el paciente con su ID generado
    }

    fun listarPacientes(): List<Paciente> {
        return repoPacientes.listarTodos()
    }
}